src/TrackedFloat.jl

```diff
@@ -25,95 +25,114 @@ end
+# floatmin2(::Type{TrackedFloat64}) = reinterpret(Float64, 0x21a0000000000000)
+
 for TrackedFloatN in (:TrackedFloat16, :TrackedFloat32, :TrackedFloat64)
 
-# FloatN is the base float type derived from TrackedFloatN
-@eval FloatN = $(Symbol("Float", string(TrackedFloatN)[end-1:end]))
+  # FloatN is the base float type derived from TrackedFloatN
+  @eval FloatN = $(Symbol("Float", string(TrackedFloatN)[end-1:end]))
 
-@eval begin
-  struct $TrackedFloatN <: AbstractTrackedFloat
-    val::$FloatN
-  end
+  @eval begin
+    struct $TrackedFloatN <: AbstractTrackedFloat
+      val::$FloatN
+    end
 
-  Base.Float64(x::$TrackedFloatN) = Float64(x.val)
-  Base.Float32(x::$TrackedFloatN) = Float32(x.val)
-  Base.Float16(x::$TrackedFloatN) = Float16(x.val)
-  Base.Int64(x::$TrackedFloatN) = Int64(x.val)
-  Base.Int32(x::$TrackedFloatN) = Int32(x.val)
-  Base.Int16(x::$TrackedFloatN) = Int16(x.val)
-
-  Base.bitstring(x::$TrackedFloatN) = bitstring(x.val)
-  Base.show(io::IO,x::$TrackedFloatN) = print(io, $TrackedFloatN,"(",string(x.val),")")
-
-  # $TrackedFloatN(x::AbstractFloat) = $TrackedFloatN(x)
-  # $TrackedFloatN(x::Integer) = $TrackedFloatN(x)
-  $TrackedFloatN(x::Rational{}) = $TrackedFloatN($FloatN(x))
-  $TrackedFloatN(x::$TrackedFloatN) = $TrackedFloatN(x.val)
-  $TrackedFloatN(x::Bool) = $TrackedFloatN($FloatN(x))
-
-  Base.promote_rule(::Type{<:Integer},::Type{$TrackedFloatN}) = $TrackedFloatN
-  Base.promote_rule(::Type{Float64},::Type{$TrackedFloatN}) = $TrackedFloatN
-  Base.promote_rule(::Type{Float32},::Type{$TrackedFloatN}) = $TrackedFloatN
-  Base.promote_rule(::Type{Float16},::Type{$TrackedFloatN}) = $TrackedFloatN
-  Base.promote_rule(::Type{Bool},::Type{$TrackedFloatN}) = $TrackedFloatN
-end
+    Base.Float64(x::$TrackedFloatN) = Float64(x.val)
+    Base.Float32(x::$TrackedFloatN) = Float32(x.val)
+    Base.Float16(x::$TrackedFloatN) = Float16(x.val)
+    Base.Int64(x::$TrackedFloatN) = Int64(x.val)
+    Base.Int32(x::$TrackedFloatN) = Int32(x.val)
+    Base.Int16(x::$TrackedFloatN) = Int16(x.val)
+
+    Base.bitstring(x::$TrackedFloatN) = bitstring(x.val)
+    Base.show(io::IO,x::$TrackedFloatN) = print(io, $TrackedFloatN,"(",string(x.val),")")
+
+    # $TrackedFloatN(x::AbstractFloat) = $TrackedFloatN(x)
+    # $TrackedFloatN(x::Integer) = $TrackedFloatN(x)
+    $TrackedFloatN(x::Rational{}) = $TrackedFloatN($FloatN(x))
+    $TrackedFloatN(x::$TrackedFloatN) = $TrackedFloatN(x.val)
+    $TrackedFloatN(x::Bool) = $TrackedFloatN($FloatN(x))
+
+    Base.promote_rule(::Type{<:Integer},::Type{$TrackedFloatN}) = $TrackedFloatN
+    Base.promote_rule(::Type{Float64},::Type{$TrackedFloatN}) = $TrackedFloatN
+    Base.promote_rule(::Type{Float32},::Type{$TrackedFloatN}) = $TrackedFloatN
+    Base.promote_rule(::Type{Float16},::Type{$TrackedFloatN}) = $TrackedFloatN
+    Base.promote_rule(::Type{Bool},::Type{$TrackedFloatN}) = $TrackedFloatN
+
+    floatmin2($TrackedFloatN) = floatmin2($FloatN)
```

Comment:
Is `floatmin2` a standard lib function? If not, I would be hesitant to add it to tracked floats operations. Ideally, we'd be logging at the level of operations called by `floatmin2`. But, if this is needed to make something work and still gives us info we need, I'm all for it.